kubernetes/dynamic.go

```diff
@@ -468,58 +474,47 @@ func (c *Client) QueryResources(ctx context.Context, selector types.ResourceSele
 
 	var resources []unstructured.Unstructured
 	for _, kind := range selector.Types {
-		client, err := c.GetClientByKind(kind)
+		client, rm, err := c.GetClientByKind(strings.TrimPrefix(kind, "Kubernetes::"))
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to get client for %s: %w", kind, err)
 		}
-		var namespaces []string
 
-		if strings.ToLower(kind) == "namespace" {
-			if name, ok := selector.ToGetOptions(); ok {
-				if selector.IsMetadataOnly() {
-					return []unstructured.Unstructured{
-						unstructured.Unstructured{
-							Object: map[string]any{
-								"kind":        "Namespace",
-								"apiVersion:": "v1",
-								"metadata": map[string]any{
-									"name": name,
-								},
-							},
-						},
-					}, nil
-				}
-				namespaces = []string{name}
-			} else {
-				namespaces = []string{""}
-			}
+		isClusterScoped := rm.Scope.Name() == meta.RESTScopeNameRoot
+
+		var namespaces []string
+		if isClusterScoped {
+			namespaces = []string{""}
```

Comment:
We seem to have dropped the selector IsMetadataOonly support which means we hit the API to get the namespace but never actually use anything besides for the name

Reply:
I've added it back but I couldn't see it being used anywhere